cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ScheduleSpecificationCodec.java

```diff
@@ -32,11 +34,48 @@
  *
  */
 public class ScheduleSpecificationCodec extends AbstractSpecificationCodec<ScheduleSpecification> {
+
+  /**
+   * Schedule Type.
+   */
+  private enum ScheduleType {
+    ORIGINAL_TIME,
```

Comment:
This is not needed if you replace Schedule with a TimeSchedule in the configurer.addSchedule()